### ISSUE TITLE
Parsing of bad self-closing tags a bit strange

### ISSUE THREAD

**Issue by Manishearth** [servo/html5ever#89]
I'm not entirely sure if this is a bug, but giving Servo the following HTML

```
<html>
<head></head>
<body>
<div id="a">
    <picture><svg><font/><source srcset='data:,b'/></svg><img src='data:,a' data-expect='data:,a'></picture>
</div>

<script type="text/javascript">
    alert(document.getElementById("a").innerHTML)
</script>

</body>
</html>
```

Gives ` <picture></picture><svg></svg><font></font><source srcset="data:,b"><font></font><img src="data:,a" data-expect="data:,a"><font></font>` in the alert

The second font tag seems wrong to me (and Firefox isn't showing it)

**Comment by Ms2ger** [comment]
This is just "we don't support foreign content", I think. The `<font/>` should be treated as `<font></font>` inside the SVG, but is being parsed as plain HTML `<font>

**Comment by Manishearth** [comment]
ah, okay. So I guess we should ignore this failure in the rustup.

**Comment by Ms2ger** [comment]
(Though it'd be good to figure out why this changed.)

**Comment by nox** [comment]
Servo now gives:

```
<picture><svg><font></font><source srcset="data:,b"></source></svg><img src="data:,a" data-expect="data:,a"></picture>
```

#160 and #161 take care of the rest of the issues with integration points.

**Comment by nox** [comment]
Fixed by #160 and #161.
